crates/cargo-winrt/src/main.rs

```diff
@@ -438,70 +437,27 @@ impl DependencyDescriptor {
     }
 }
 
-fn try_download(url: String, recursion_amount: u8) -> anyhow::Result<Vec<u8>> {
-    if recursion_amount == 0 {
-        bail!(Error::DownloadError(
-            anyhow::anyhow!("Too many redirects").into(),
-        ));
-    }
+fn try_download(url: String) -> anyhow::Result<Vec<u8>> {
     let mut handle = Easy::new();
-    handle
-        .url(&url)
-        .map_err(|e| Error::DownloadError(e.into()))?;
-    let status = &mut None;
+    handle.url(&url)?;
```

Comment:
Curl errors should be [mapped](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err) to `DownloadError`.